lib/create-app.js

```diff
@@ -143,9 +144,32 @@ function initViews (app, configPath) {
 function initWebId (argv, app, ldp) {
   config.ensureWelcomePage(argv)
 
-  // Use session cookies
+  // Store in a session cookie whether the user is logged in
+  // (for direct browsing by people only)
   const useSecureCookies = argv.webid  // argv.webid forces https and secure cookies
-  app.use(session(sessionSettings(useSecureCookies, argv.host)))
+  const sessionHandler = session(sessionSettings(useSecureCookies, argv.host))
+  app.use((req, res, next) => {
+    sessionHandler(req, res, () => {
+      // Reject cookies from third-party applications.
+      // Otherwise, when a user is logged in to their Solid server,
+      // any third-party application could perform authenticated requests
+      // without permission by including the credentials set by the Solid server.
+      const origin = req.headers.origin
+      const userId = req.session.userId
+      console.log('check')
+      if (origin && userId) {
+        const userIdHost = userId.replace(/([^:/])\/.*/, '$1')
```

Comment:
I think this might grab the path, as well as the host? so, it'll be like `ruben.databox.me/profile/card`, yeah?